pyrobosim/pyrobosim/navigation/astar_grid.py

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/env python3
+import time
+import math
+import warnings
+from queue import PriorityQueue
+from pyrobosim.utils.pose import Pose
+from pyrobosim.utils.motion import Path
+from pyrobosim.navigation.occupancy_grid import occupancy_grid_from_world
+
+
+class Node:
+    """A node in the A* path"""
+
+    def __init__(self, x, y, g, h, parent=None) -> None:
+        self.x = x
+        self.y = y
+        self.g = g  # Cost to reach this node from start
+        self.h = h  # Approximate cost to goal from this node
+        self.parent = parent
+
+    @property
+    def f(self):
+        """Total cost to goal via this node"""
+        return self.g + self.h
+
+    def __eq__(self, other):
+        return self.x == other.x and self.y == other.y
+
+    def __le__(self, other):
+        return self.f <= other.f
+
+    def __ge__(self, other):
+        return self.f >= other.f
+
+    def __lt__(self, other):
+        return self.f < other.f
+
+    def __gt__(self, other):
+        return self.f > other.f
+
+    def __ne__(self, other):
+        return self.x != other.x or self.y != other.y
+
+    def __repr__(self) -> str:
+        return f"(Node -> {self.x}, {self.y}), Cost : {self.f}\n"
+
+
+class AStarGridPlanner:
+    """Occupancy grid based A* planner"""
+
+    def __init__(
+        self,
+        world,
+        resolution=0.05,
+        inflation_radius=0.0,
+        distance_metric="Manhattan",
```

Comment:
I would default to Euclidean as that one is a correct admissible heuristic with/without diagonal moves enabled.

Also for consistency, I think these should be lowercase : `manhattan` and `euclidean`